### PR TITLE
chore(authenticator): add authenticator widget tests

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:*

*Description of changes:*
- move shared authenticator test utils to new package, amplify_authenticator_test
- add widget tests for sign in/up happy path and form validation

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_test/lib/src/stubs/amplify_auth_cognito_stub.dart [hunk 1]
```diff
@@ -313,29 +335,44 @@ class AmplifyAuthCognitoStub extends AuthPluginInterface
...
 
-  const _MockUser({
+  factory MockCognitoUser({
+    required String username,
+    required String password,
+    String? email,
+    String? phoneNumber,
+  }) {
+    return MockCognitoUser._(
+      sub: Random().nextInt(10000).toString(),
```
Comment (dnys1) [inline packages/amplify_test/lib/src/stubs/amplify_auth_cognito_stub.dart:352]: `uuid()` would be higher fidelity - not sure it matters though

File: packages/authenticator/amplify_authenticator/example/macos/Runner.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -400,8 +418,10 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_ENTITLEMENTS = Runner/DebugProfile.entitlements;
+				"CODE_SIGN_IDENTITY[sdk=macosx*]" = "Apple Development";
 				CODE_SIGN_STYLE = Automatic;
 				COMBINE_HIDPI_IMAGES = YES;
+				DEVELOPMENT_TEAM = R9KJJV3A3U;
```
Comment (dnys1) [inline packages/authenticator/amplify_authenticator/example/macos/Runner.xcodeproj/project.pbxproj:424]: Remove

File: packages/authenticator/amplify_authenticator/example/macos/Runner/Release.entitlements [hunk 3]
```diff
@@ -6,5 +6,9 @@
 	<true/>
 	<key>com.apple.security.network.client</key>
 	<true/>
+	<key>keychain-access-groups</key>
```
Comment (dnys1) [inline packages/authenticator/amplify_authenticator/example/macos/Runner/Release.entitlements:9]: Can we align with https://github.com/aws-amplify/amplify-flutter/pull/2396?

I found this to be a nicer experience than requiring a development account right off the bat.

File: packages/authenticator/amplify_authenticator/test/sign_in_test.dart [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_authenticator_test/amplify_authenticator_test.dart';
+import 'package:amplify_test/amplify_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  LiveTestWidgetsFlutterBinding();
```
Comment (dnys1) [inline packages/authenticator/amplify_authenticator/test/sign_in_test.dart:20]:
```suggestion
  LiveTestWidgetsFlutterBinding.ensureInitialized();
```

Comment (dnys1) [reply packages/authenticator/amplify_authenticator/test/sign_in_test.dart:20]: Should all tests have this?

File: packages/authenticator/amplify_authenticator/test/sign_in_test.dart [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      );
+
+      await tester.pumpWidget(
+        MockAuthenticatorApp(
+          authPlugin: AmplifyAuthCognitoStub(users: [testUser]),
+        ),
+      );
+      await tester.pumpAndSettle();
+
+      // final signUpPage = SignUpPage(tester: tester);
```
Comment (dnys1) [inline packages/authenticator/amplify_authenticator/test/sign_in_test.dart:38]: Remove

File: packages/authenticator/amplify_authenticator/test/sign_in_test.dart [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      await signInPage.enterUsername(testUser.email!);
+
+      // And I type my password
+      await signInPage.enterPassword(testUser.password);
+
+      // And I click the "Sign In" button
+      await signInPage.submitSignIn();
+
+      // Then I see "Sign out"
+      await confirmSignUpPage.expectAuthenticated();
```
Comment (dnys1) [inline packages/authenticator/amplify_authenticator/test/sign_in_test.dart:52]: This should work, no?

```suggestion
      await signInPage.expectAuthenticated();
```

File: packages/authenticator/amplify_authenticator/test/sign_up_form_test.dart [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+          expect(passwordFieldErrorLine3, findsOneWidget);
+          expect(passwordFieldErrorLine4, findsOneWidget);
+        },
+      );
+    });
+  });
+}
+
+const passwordReqConfig = AmplifyConfig(
+  auth: AuthConfig(
```
Comment (dnys1) [inline packages/authenticator/amplify_authenticator/test/sign_up_form_test.dart:144]: For future ref, you can use `AuthConfig.cognito` to make this easier

File: packages/authenticator/amplify_authenticator/test/sign_up_test.dart [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_authenticator_test/amplify_authenticator_test.dart';
+import 'package:amplify_test/amplify_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  LiveTestWidgetsFlutterBinding();
```
Comment (dnys1) [inline packages/authenticator/amplify_authenticator/test/sign_up_test.dart:20]: For consistency

```suggestion
  LiveTestWidgetsFlutterBinding.ensureInitialized();
```

**Review by dnys1 — APPROVED** [review]
Minor comments - looks good though!

File: packages/authenticator/amplify_authenticator/example/macos/Runner/Release.entitlements (hunk 3 above)
Comment (Jordan-Nelson) [reply packages/authenticator/amplify_authenticator/example/macos/Runner/Release.entitlements:9]: Yeah. I agree that is probably better. You will occasionally have to allow access, which is unfortunately done on a key by key basis, so you have to allow access ~11 times by entering your password. Still probably better than the alternative for most people.

File: packages/authenticator/amplify_authenticator/test/sign_in_test.dart (hunk 6 above)
Comment (Jordan-Nelson) [reply packages/authenticator/amplify_authenticator/test/sign_in_test.dart:52]: Yup. Copy/paste mistake. Will update.

File: packages/authenticator/amplify_authenticator/test/sign_up_form_test.dart (hunk 7 above)
Comment (Jordan-Nelson) [reply packages/authenticator/amplify_authenticator/test/sign_up_form_test.dart:144]: Oh, nice! I will update this

**Review by Equartey — APPROVED** [review]

**Pull request merged**